odoo/tools/misc.py

```diff
@@ -256,17 +256,16 @@ def file_open(name: str, mode: str = "r", filter_ext: tuple[str, ...] = (), env:
     :raise ValueError: if the file doesn't have one of the supported extensions (`filter_ext`)
     """
     path = file_path(name, filter_ext=filter_ext, env=env)
-    if os.path.isfile(path):
-        if 'b' not in mode:
-            # Force encoding for text mode, as system locale could affect default encoding,
-            # even with the latest Python 3 versions.
-            # Note: This is not covered by a unit test, due to the platform dependency.
-            #       For testing purposes you should be able to force a non-UTF8 encoding with:
-            #         `sudo locale-gen fr_FR; LC_ALL=fr_FR.iso8859-1 python3 ...'
-            # See also PEP-540, although we can't rely on that at the moment.
-            return open(path, mode, encoding="utf-8")
-        return open(path, mode)
-    raise FileNotFoundError("Not a file: " + name)
+    encoding = None
+    if 'b' not in mode:
+        # Force encoding for text mode, as system locale could affect default encoding,
+        # even with the latest Python 3 versions.
+        # Note: This is not covered by a unit test, due to the platform dependency.
+        #       For testing purposes you should be able to force a non-UTF8 encoding with:
+        #         `sudo locale-gen fr_FR; LC_ALL=fr_FR.iso8859-1 python3 ...'
+        # See also PEP-540, although we can't rely on that at the moment.
+        encoding = "utf-8"
+    return open(path, mode, encoding=encoding)
```

Comment:
the function now can be used to create new files (`mode='w'), I don't know if we want to forbid that?

Reply:
This should not be an issue as this function is only used internally and the default mode is "r".
Alternatively, we could add `if {'w', 'x', 'a'}.intersection(mode) and not isfile(path): raise FileNotFoundError(...)` before opening.